### PR TITLE
Replace unwraps in compiler glyph embedding with diagnostics

### PR DESCRIPTION
**Pull request by tronical**
As discussed with Olivier, fatal diagnostics are better than a panicing compiler.

### CODE REVIEW

File: internal/compiler/passes/embed_glyphs.rs [hunk 1]
```diff
@@ -43,9 +46,23 @@ pub fn embed_glyphs<'a>(
...
+        for custom_size_str in sizes_str.split(',') {
+            let custom_size = if let Ok(custom_size) = custom_size_str
+                .parse::<f64>()
+                .map(|size_as_float| (size_as_float * scale_factor) as i16)
+            {
+                custom_size
+            } else {
+                diag.push_error(
+                    format!(
+                        "Invalid font size {} specified in `SLINT_FONT_SIZES`",
```
Comment (ogoffart) [inline internal/compiler/passes/embed_glyphs.rs:58]:
```suggestion
                        "Invalid font size '{}' specified in `SLINT_FONT_SIZES`",
```

File: internal/compiler/passes/embed_glyphs.rs [hunk 2]
```diff
@@ -145,67 +163,106 @@ pub fn embed_glyphs<'a>(
...
             .with_face_data(face_id, |font_data, face_index| {
-                let font = fontdue::Font::from_bytes(
-        font_data,
-        fontdue::FontSettings { collection_index: face_index, scale: 40. },
-    )
-    .expect("internal error: fontdb returned a font that ttf-parser/fontdue could not parse");
+                let fontdue_font = if let Ok(fontdue_font) = fontdue::Font::from_bytes(font_data, fontdue::FontSettings { collection_index: face_index, scale: 40. }) {
+                    fontdue_font
+                } else {
+                    diag.push_error("internal error: mfontdb returned a font that ttf-parser/fontdue could not parse".to_string(), &generic_diag_location);
```
Comment (ogoffart) [inline internal/compiler/passes/embed_glyphs.rs:230]: Could add the actual error here.

File: internal/compiler/passes/embed_glyphs.rs [hunk 3]
```diff
@@ -145,67 +163,106 @@ pub fn embed_glyphs<'a>(
...
                     characters_seen.iter().cloned(),
                     &fallback_fonts,
-                )
-            })
-            .unwrap();
+                ).into()
+            }) {
+            maybe_font
+        } else {
+            diag.push_error("internal error: face_id of selected font is unknown to fontdb".to_string(), &generic_diag_location);
```
Comment (ogoffart) [inline internal/compiler/passes/embed_glyphs.rs:256]: (also valid for other error messages) Is there a way to specify which font is incorrect? 
As an user, if i'm getting this error, i don't really know what font went wrong.

**Review by ogoffart — APPROVED** [review]

File: internal/compiler/passes/embed_glyphs.rs (hunk 2 above)
Comment (tronical) [reply internal/compiler/passes/embed_glyphs.rs:230]: Good idea!

File: internal/compiler/passes/embed_glyphs.rs (hunk 3 above)
Comment (tronical) [reply internal/compiler/passes/embed_glyphs.rs:256]: Hmmmn, maybe. This particular situation really can't quite happen, but for debugging purposes I can try to enhance this and others.

**Review by ogoffart — APPROVED** [review]

**Pull request merged**